protokt-codegen/src/main/resources/renderers.stg

```diff
@@ -118,15 +118,18 @@ deserializeValue(p) ::= <%
     <endif>
 %>
 
-requireNotNullMessage(p) ::= <%
-    "<p.name> specified nonnull with (protokt.<if (p.oneOf)>oneof<else>property<endif>).non_null but was null"
-%>
+requireNotNullMessage(p) ::= <<
+requireNotNull(<p.name>) {
+    StringBuilder("<p.name>")
```

Comment:
why do we need a stringbuilder?

Reply:
If we just use concatenation, the compiler will "optimize" and do the concatenation at compile-time, which we don't want. This whole PR is an optimization geared towards Android - attempting to avoid creation of rarely needed long string literals in generated code, which tends to proliferate.

Reply:
In an earlier commit this is a function exposed by the runtime, but I thought it's probably better not to expose a method just for this. Maybe this can be pulled into a locally generated method?

Reply:
The Kotlin compiler generates another string for its intrinsics:
```java
var26 = (new StringBuilder("socketAddress")).append(" specified nonnull with (protokt.property).non_null but was null");
Intrinsics.checkExpressionValueIsNotNull(var26, "StringBuilder(\"socketAdd…).non_null but was null\")");
```

But this at least allows the protokt string to be reused.

Reply:
you could pull the shared parts of errors messages into runtime and generate something like

```
"uuid ${Errors.IS_A_MISSING_WRAPPED_PRIMITIVE}"
```